core/src/main/java/io/confluent/rest/ApplicationServer.java

```diff
@@ -603,8 +605,16 @@ private static ThreadPool createThreadPool(RestConfig config) {
     log.info("Initial capacity {}, increased by {}, maximum capacity {}.",
             initialCapacity, growBy, maxCapacity);
 
+    if (initialCapacity > maxCapacity) {
+      threadPoolRequestQueueCapacity = Math.max(initialCapacity, 8) * 1024;
+      log.warn("request.queue.capacity is less than request.queue.capacity.init, invalid config. "
+          + "Setting request.queue.capacity to at least 1024 * request.queue.capacity.init.");
```

Comment:
I think this violates the [principle of least surprise](https://en.wikipedia.org/wiki/Principle_of_least_astonishment).

It's unintuitive and I think not very practical to have the `maxCapacity` be set to 10 if both it and the initial capacity are configured to be 10, but to more than 10k if it is configured to be 9 while the initial capacity is configured to be 10. Let's just set it to `initialCapacity` in that case - note that there's a specific intent here, unlike the `null` queue case where this very generous sizing originates from.